### PR TITLE
Added blocks filter to SpeedMine module

### LINKED ISSUE #2773: Add a whitelist/blacklist to speedmine
### Describe the feature

Add a whitelist to speed mine to only speed mine certain blocks, such as obsidian.

### Before submitting a suggestion

- [X] This feature doesn't already exist in the client. (I have checked every module and their settings on the **latest dev build**)

- [X] This wasn't already suggested. (I have searched suggestions on GitHub and read the FAQ)

### PR DESCRIPTION
**Pull request by DesiCow**
Whitelist/Blacklist certain blocks to be speed mined.
Closes #2773

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java [hunk 1]
```diff
@@ -7,20 +7,38 @@
...
+
 import static net.minecraft.entity.effect.StatusEffects.HASTE;
 
 public class SpeedMine extends Module {
     private final SettingGroup sgGeneral = settings.getDefaultGroup();
 
+    private final Setting<List<Block>> blocks = sgGeneral.add(new BlockListSetting.Builder()
+            .name("blocks")
+            .description("Selected blocks.")
+            .filter(block -> block.getHardness() > 0)
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java:31]: Why this filter?

Comment (RickyTheRacc) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java:31]: Some anticheats will only allow speedmine to work on combat blocks like obsidian, so you'd have to turn it on and off when you want to mine normal blocks like stone

**Comment by I-Got-You-Dead** [comment]
Based pr ngl

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java (hunk 1 above)
Comment (DesiCow) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/player/SpeedMine.java:31]: So you don't get unnecessary blocks like end portal, bedrock, nether portal, etc. in blocks list

**Pull request merged**